File: InfixPrefixPostfixExp/PrefixToPostfix/preToPost.py
class Solution:
    def checkOperator(self, op):
        if op == "+" or op == "%" or op == "-" or op == "*" or op == "/" or op == "^":
            return True
        return False

    def solve(self, val1, op, val2):
        return val1 + val2 + op

    def preToPost(self, s):
        st = []

        for i in range(len(s) - 1, -1, -1):
            term = s[i]

            if not self.checkOperator(term):
                st.append(term)
            else:
                val1 = st.pop()
                val2 = st.pop()

                res = self.solve(val1, term, val2)
                st.append(res)

        return st[-1]

File: InfixPrefixPostfixExp/PrefixToPostfix/preToPost.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
  public:
    bool checkOperator(string op){
       if(op == "+" || op == "%" || op == "-" || op == "*" || op == "/" || op == "^")return true;
       return false;
    }
    string solve(string val1, string op, string val2){
        string ans = "";
        ans += val1;
        ans += val2;
        ans += op;
        return ans;
    }
    string preToPost(string &s) {
        // code here
        stack<string>st;
        int n = s.size();
        for(int i=n-1;i>=0;i--){
            string term = string(1,s[i]);
            if(!checkOperator(term)){
                st.push(term);
            }else{
                string val1 = st.top();
                st.pop();
                string val2 = st.top();
                st.pop();
                string res = solve(val1,term,val2);
                st.push(res);
            }
        }
        return st.top();
    }
};